File: AdvStudCompat.java

/***
 * 
 * This class is designed to represent the compatibility between a 
 * student and an advisor
 *
*/
public class AdvStudCompat{
	// a particular student 
	private Student student;

	// a particular advisor 
	private Advisor advisor;

	// the compatibility between the student and the advisor 
	private float compatibility;

	/**
	 * @param : 
	 *		 - a student
	 *		 - an advisor 
	 * 		 - the compatibility between the student and the advisor 
	 * 
	 * @return - instantiates a compatibility class object
	*/
	public AdvStudCompat(Student student, Advisor advisor, float compatibility){
		this.student = student;
		this.advisor = advisor;
		this.compatibility = compatibility;
	}

	/**
	 * @pre - an existing compatibility object 
	 * 
	 * @param - none
	 *
	 * @return the compatibility between the student and the advisor 
	*/
	public float getCompatibility(){
		return compatibility;
	}

	/**
	 * @pre - an existing compatibility object 
	 * 
	 * @param - none
	 *
	 * @return the advisor to this compatibility object 
	*/
	public Advisor getAdvisor(){
		return advisor;
	}

	/**
	 * @pre - an existing compatibility object 
	 * 
	 * @param - none
	 *
	 * @return the student to this compatibility object 
	*/
	public Student getStudent(){
		return student;
	}

	
}

File: javascript/AdvStudCompat.js
/**
 * Construct an AdvisorStudentCompat Object instance.
 *
 * @pre - valid student, advisor and compatibility objects
 * @post - constructs a AdvStudCompat object used by the runner class to match 
 *         students to advisors
 *        
 * @params:
 *         student - student(object) being matched
 *         advisor - advisor(object) being matched
 *         compatibility - float compatibility between this student and this advisor
 * 
 * @class
 */
var AdvStudCompat = (function () {
    function AdvStudCompat(student, advisor, compatibility) {
    	if (this.student === undefined){
        this.student = null;
      }
      if (this.advisor === undefined){
        this.advisor = null;
      }
      if (this.compatibility === undefined){
        this.compatibility = null;
      }
    	this.student = student;
      this.advisor = advisor;
      this.compatibility = compatibility;
    };

  /**
   * @pre - an existing compatibility object 
   * 
   * @param - none
   *
   * @return the float compatibility between the student and the advisor 
  */
  AdvStudCompat.prototype.getCompatibility = function(){
    return this.compatibility;
  };

  /**
   * @pre - an existing compatibility object 
   * 
   * @param - none
   *
   * @return the advisor to this compatibility object 
  */
  AdvStudCompat.prototype.getAdvisor = function(){
    return this.advisor;
  };

  /**
   * @pre - an existing compatibility object 
   * 
   * @param - none
   *
   * @return the student to this compatibility object 
  */
  AdvStudCompat.prototype.getStudent = function(){
    return this.student;
  };
  return AdvStudCompat;
}());
AdvStudCompat["__class"] = "AdvStudCompat";
